diff --git a/dist/index.js b/dist/index.js
index 3c7b76d6e49df3c8774398e4d6f24ada314da005..cb273f2950dcbf3f8797e1706711f7aabd9fb4f1 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -47,7 +47,7 @@ $82dc8d030dec7549$export$8c6ed5c666ac1360.displayName = 'Slot';
     const { children: children , ...slotProps } = props;
     if (/*#__PURE__*/ $dAvBt$react.isValidElement(children)) return /*#__PURE__*/ $dAvBt$react.cloneElement(children, {
         ...$82dc8d030dec7549$var$mergeProps(slotProps, children.props),
-        ref: forwardedRef ? $dAvBt$radixuireactcomposerefs.composeRefs(forwardedRef, children.ref) : children.ref
+        ref: forwardedRef ? $dAvBt$radixuireactcomposerefs.composeRefs(forwardedRef, children.props.ref) : children.ref
     });
     return $dAvBt$react.Children.count(children) > 1 ? $dAvBt$react.Children.only(null) : null;
 });
diff --git a/dist/index.mjs b/dist/index.mjs
index b11398c2c45bdf5b48a5cb7caac55bac228002a1..6a044c9143bc359ca9abb9e7d30468fbbdd97af2 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -37,7 +37,7 @@ $5e63c961fc1ce211$export$8c6ed5c666ac1360.displayName = 'Slot';
     const { children: children , ...slotProps } = props;
     if (/*#__PURE__*/ $9IrjX$isValidElement(children)) return /*#__PURE__*/ $9IrjX$cloneElement(children, {
         ...$5e63c961fc1ce211$var$mergeProps(slotProps, children.props),
-        ref: forwardedRef ? $9IrjX$composeRefs(forwardedRef, children.ref) : children.ref
+        ref: forwardedRef ? $9IrjX$composeRefs(forwardedRef, children.props.ref) : children.ref
     });
     return $9IrjX$Children.count(children) > 1 ? $9IrjX$Children.only(null) : null;
 });
