diff --git a/dist/index.js b/dist/index.js
index 8d89ba7feaaa4cfec92a78feebcf5a8526ae8552..9e5fdaa2da1be206b0cfefdd5a9d5db076cd62d2 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -27,7 +27,7 @@ const $a2fa0214bb2735a1$export$99c2b779aa4e8b8b = (props)=>{
     const child = typeof children === 'function' ? children({
         present: presence.isPresent
     }) : $fnLeV$react.Children.only(children);
-    const ref = $fnLeV$radixuireactcomposerefs.useComposedRefs(presence.ref, child.ref);
+    const ref = $fnLeV$radixuireactcomposerefs.useComposedRefs(presence.ref, child.props.ref);
     const forceMount = typeof children === 'function';
     return forceMount || presence.isPresent ? /*#__PURE__*/ $fnLeV$react.cloneElement(child, {
         ref: ref
diff --git a/dist/index.mjs b/dist/index.mjs
index 1a5bc83da85688082abb70f49ebee4180e550556..202105c936138c86d5184b9c542393a8675c60a7 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -22,7 +22,7 @@ const $921a889cee6df7e8$export$99c2b779aa4e8b8b = (props)=>{
     const child = typeof children === 'function' ? children({
         present: presence.isPresent
     }) : $iqq3r$Children.only(children);
-    const ref = $iqq3r$useComposedRefs(presence.ref, child.ref);
+    const ref = $iqq3r$useComposedRefs(presence.ref, child.props.ref);
     const forceMount = typeof children === 'function';
     return forceMount || presence.isPresent ? /*#__PURE__*/ $iqq3r$cloneElement(child, {
         ref: ref
